### PR TITLE
v2.3 Migration Guide

### PR DESCRIPTION
**Pull request by kbrizzle**

### CODE REVIEW

File: runbooks/MIGRATION_v2.3.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+## Checklist
+
+0. [Pre-upgrade action items](#pre-upgrade-action-items)
+2. [Enter *settle only* mode, and settle all accounts on all markets / vaults](#enter-settle-only-mode-and-settle-all-accounts-on-all-markets--vaults)
+3. [Upgrade to v2.3](#upgrade-to-v23)
+
+## Pre-upgrade action items
+
+- Deploy v2.3 protocol implementations
+- Deploy fresh v2.3 Pyth oracle instance
```
Comment (arjun-io) [inline runbooks/MIGRATION_v2.3.md:12]: We should also note that we need to deploy the new gas oracles and multicall4

File: runbooks/MIGRATION_v2.3.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+### Oracle
+
+`IOracleProvider.at()` now has a new return type, returning a `OracleReceipt` in addition to the `OracleVersion`.
+
+To ensure that the first settlement after the switchover processes correctly, we must settle up the latest global version after entering settle only mode.
+
+see: https://github.com/equilibria-xyz/perennial-v2/pull/373.
+
+## Upgrade to v2.3
```
Comment (arjun-io) [inline runbooks/MIGRATION_v2.3.md:47]: Are there any backfill values we need to set here? I think the oracle names will be handled on init but there was something about the ID mappings?

File: runbooks/MIGRATION_v2.3.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+### Position Migration
+
+- Call `.migrate()` on each market to migrate the storage pattern of the global `Position` of each market.
+
+see: https://github.com/equilibria-xyz/perennial-v2/pull/424
+
+### Update Risk / Market / Protocol Parameters to new format
+
+All parameter sets have a new format. Each needs to be updated for all markets.
+- RiskParameter adds `minScale`, see: https://github.com/equilibria-xyz/perennial-v2/pull/362.
```
Comment (arjun-io) [inline runbooks/MIGRATION_v2.3.md:65]: `minScale` should be part of ProtocolParameter

Additionally, do we need to do anything for the change in the risk parameter's maker adiabatic fee to `NoOp`? We also need to update each market's MarketParameter with the maker/taker fees

Comment (kbrizzle) [reply runbooks/MIGRATION_v2.3.md:65]: Nothing additional to what's the in guide (just resetting everything), but I added a lot more notes into this section for clarity.

File: runbooks/MIGRATION_v2.3.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+- [Update Risk / Market / Protocol Parameters to new format](#update-risk--market--protocol-parameters-to-new-format)
+- [Update sub-oracles to v2.3-based oracles](#update-sub-oracles-to-v23-based-oracles-implementations)
+
+### Position Migration
+
+- Call `.migrate()` on each market to migrate the storage pattern of the global `Position` of each market.
+
+see: https://github.com/equilibria-xyz/perennial-v2/pull/424
+
+### Update Risk / Market / Protocol Parameters to new format
```
Comment (arjun-io) [inline runbooks/MIGRATION_v2.3.md:64]: We also need to `register` the market to it's oracle

File: runbooks/MIGRATION_v2.3.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+All parameter sets have a new format. Each needs to be updated for all markets.
+- Oracle
+  - OracleParameter
+  - KeeperOracleParameter (per sub-oracle)
```
Comment (arjun-io) [inline runbooks/MIGRATION_v2.3.md:69]: I think the top level KeeperFactory needs to be updated which propagates the parameter to each KeeperOracle?

File: runbooks/MIGRATION_v2.3.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+- MarketParameter
+  - removes `.oracleFee` and `.settlementFee`, see: https://github.com/equilibria-xyz/perennial-v2/pull/373.
+  - replaces `.positionFee` with `.makerFee` and `.takerFee`, see: https://github.com/equilibria-xyz/perennial-v2/pull/291.
+- RiskParameter removes `.makerFee.adiabaticFee`, see: https://github.com/equilibria-xyz/perennial-v2/pull/317.
+- ProtocolParameter adds `minScale`, see: https://github.com/equilibria-xyz/perennial-v2/pull/362.
+- VaultParameter adds `minDeposit`, see: https://github.com/equilibria-xyz/perennial-v2/pull/373.
+
+### Update sub-oracles to v2.3-based oracles implementations
+
+A new PythOracle implementation will be deployed with this release. A standard sub-oracle update is required on each market's oracle to transition to the new sub-oracles.
```
Comment (arjun-io) [inline runbooks/MIGRATION_v2.3.md:87]: "New PythFactory and CryptexFactory implementations..."

**Review by arjun-io — APPROVED** [review]

**Pull request merged**